service/api_task.go

```diff
@@ -322,7 +323,17 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, currentHost *host.Host)
 	for taskQueue.Length() != 0 {
 		var queueItem *model.TaskQueueItem
 		var err error
-		switch currentHost.Distro.PlannerSettings.Version {
+		d, err := distro.FindOne(distro.ById(currentHost.Distro.Id))
+		if err != nil {
+			grip.Critical(message.WrapError(err, message.Fields{
+				"distro":  currentHost.Distro.Id,
+				"host":    currentHost.Id,
+				"message": "problem finding distro",
+				"spec":    spec,
+			}))
+			return nil, errors.Wrapf(err, "problem finding distro %s", currentHost.Distro.Id)
```

Comment:
this is going to happen when we delete a distro. I think in this case we should fall back to using the embeded distro with a warning? 

also maybe just error, when this happens it's going to happen a lot and isn't really a big.

Reply:
done